extensions/fairifier-rdf-extension/dtls/fairifier/DetectFileFormatCommand.java

```diff
@@ -1,145 +1,153 @@
-        
-        this.parsers = new RDFParser[]{this.rdfParserntriples, this.rdfParserttl};
+    public void doPost(HttpServletRequest req, HttpServletResponse res)
+            throws ServletException, IOException {
+
+        this.parsers = new RDFParser[] {this.rdfParserntriples, this.rdfParserttl,
+                this.rdfParserrdfxml, this.rdfParserJsonld, this.rdfParserNQuads,
+                this.rdfParserJson, this.rdfParserTrix, this.rdfParserTrig};
         FileItemFactory factory = new DiskFileItemFactory();
 
         // Create a new file upload handler
         ServletFileUpload upload = new ServletFileUpload(factory);
         List<FileItem> items = null;
         String baseuri = null;
         String filename = null;
-        InputStream in = null;
+        InputStream[] in = new InputStream[this.parsers.length];
         String format = null;
         String url = null;
-        
-        try{
+
+        try {
             items = upload.parseRequest(req);
-            for(FileItem item:items){
-                if(item.getFieldName().equals("baseuri")){
+            for (FileItem item : items) {
+                if (item.getFieldName().equals("baseuri")) {
                     baseuri = item.getString();
-                }else if (item.getFieldName().equals("file_source")) {
+                } else if (item.getFieldName().equals("file_source")) {
                     url = item.getString();
-                }else if(item.getFieldName().equals("file_upload")){
+                } else if (item.getFieldName().equals("file_upload")) {
                     filename = item.getName();
-                    in = item.getInputStream();
+                    for (int i = 0; i < in.length; i++) {
+                        StringWriter writer = new StringWriter();
+                        in[i] = new BufferedInputStream(item.getInputStream());
+
+                    }
                 }
             }
-        }catch(FileUploadException ex){
+        } catch (FileUploadException ex) {
             respondException(res, ex);
         }
+        StringBuffer response = new StringBuffer();;
 
         if ((url != null) && !url.trim().equals("") && !url.trim().equals("url")) {
             URL obj = new URL(url);
             HttpURLConnection con = (HttpURLConnection) obj.openConnection();
             BufferedReader inr = new BufferedReader(new InputStreamReader(con.getInputStream()));
             String inputLine;
-            StringBuffer response = new StringBuffer();
 
             while ((inputLine = inr.readLine()) != null) {
-                    response.append(inputLine);
+                response.append(inputLine);
             }
             inr.close();
-            in = new ByteArrayInputStream(response.toString().getBytes());
+            for (int i = 0; i < in.length; i++) {
+                in[i] = new BufferedInputStream(
+                        new ByteArrayInputStream(response.toString().getBytes("UTF8")));
+            }
```

Comment:
I get that this (entire) block tries to determine the format for an external file, but there are other ways to achieve that. Namely, using a combination of [`RDFFormat.getAcceptParams`](http://docs.rdf4j.org/javadoc/latest/org/eclipse/rdf4j/rio/RDFFormat.html#getAcceptParams-java.lang.Iterable-boolean-org.eclipse.rdf4j.rio.RDFFormat-), [`HttpUrlConnection. addRequestProperty`](http://docs.oracle.com/javase/8/docs/api/java/net/URLConnection.html#addRequestProperty-java.lang.String-java.lang.String-), and [`Rio.getParserFormatForMIMEType`](http://docs.rdf4j.org/javadoc/latest/org/eclipse/rdf4j/rio/Rio.html#getParserFormatForMIMEType-java.lang.String-). Ping me for more details.